### PR TITLE
Replace Yargs with Sade

### PR DESCRIPTION
**Pull request by swojit**
Fixes https://github.com/uber-web/framework/issues/568 
Improves speed by 15-20x

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -10,12 +10,6 @@
...
-  .run(process.argv.slice(2).join(' '))
-  .catch(e => {
-    // eslint-disable-next-line no-console
-    console.error(e.message);
-    if (e.yargs) e.yargs.showHelp();
-    // Rethrow so that there's a non-zero exit code
-    throw e;
-  });
+require('./cli-runner').run(
+  process.argv.map(x => x.split(' ')).reduce((x, y) => x.concat(y))
```
Comment (lhorie) [inline bin/cli.js]: Why is this needed?

Comment (swojit) [reply bin/cli.js]: It is because of the test files. They pass the arguments like 'dir --test` instead of tokenizing them which breaks sade.

Comment (lhorie) [reply bin/cli.js]: I think we should pass Array<string> from the tests to match parameter types under regular usage. As it stands, I'm not sure it does the right thing if we pass something like `--dir="/Program Files/something"`

Comment (swojit) [reply bin/cli.js]: It does the right thing. But yes, I was wondering if I should refactor the tests or refactor the cli code for this check. I am pushing another commit with the changes.

File: bin/cli-runner.js [hunk 2]
```diff
@@ -12,32 +12,27 @@ process.on('unhandledRejection', e => {
...
-      value: instance,
-    });
-    return Promise.reject(error);
-  } else {
-    // $FlowFixMe
-    return require(cmdPath).run(instance.argv);
+  const data = require('../commands/index.js');
+  const instance = sade('fusion').version(require('../package.json').version);
+  for (const [command, metadata] of Object.entries(data)) {
+    if (metadata instanceof Object) {
```
Comment (angus-c) [inline bin/cli-runner.js]: is this what you want? This will return true for anything that is not a primitive, null or undefined.

**Comment by swojit** [comment]
I needed this for Flow.

File: bin/cli-runner.js (hunk 2 above)
Comment (swojit) [reply bin/cli-runner.js]: I needed this for flow.
